src-cpp/rdkafkacpp.h

```diff
@@ -715,7 +715,61 @@ class RD_EXPORT OffsetCommitCb {
   virtual ~OffsetCommitCb() { }
 };
 
+/**
+* @brief SSL certificate verify callback class
+*/
+class RD_EXPORT CertVerifyCb {
+public:
+    /**
+    * @brief Set SSL certificate verification callback.
+    *
+    * The verification of the broker certificate will be processed
+    * by this callback.
+    * @returns true if the SSL certificate is successfully verified otherwise false.
+    */
+    virtual bool cert_verify_cb(char* cert, size_t len) = 0;
+
+    virtual ~CertVerifyCb() {}
+};
 
+/**
+* @brief Certificate retrieve callback class
+*/
+class RD_EXPORT CertRetrieveCb {
+public:
+    /** @brief Type of the retrieve certificate callback */
+    enum Type {
+        CERTIFICATE_PUBLIC_KEY,
+        CERTIFICATE_PRIVATE_KEY,
+        CERTIFICATE_PRIVATE_KEY_PASS
+    };
+
+    /**
+    * @brief Set certificate retrieve callback.
+    *
+    * The retrieval of the cetificate specified by the type will be returned by this callback.
+    * This is used to get the certificates required to create the SSL connection.
+    *
+    * @remark upon compleation the buffer must point to a valid buffer.
+    *
+    * @param buffer must point to valid data upon compleation.  It is the responsibility of the
+    *   implementation to ensure that the data is valid until the class is destructed.
+    *
+    * @param type the type of buffer which will be retrieved by the callback.
+    *   CERTIFICATE_PUBLIC_KEY
+    *   CERTIFICATE_PRIVATE_KEY
+    *   CERTIFICATE_PRIVATE_KEY_PASS
+    *
+    * @remark When type is CERTIFICATE_PUBLIC_KEY or CERTIFICATE_PRIVATE_KEY then upon return the buffer
+    * must point to the certificate.  When type is For CERTIFICATE_PRIVATE_KEY_PASS then buffer
+    * must point to a narrow string containing the password of the private key.
+    *
+    * @returns the number of bytes in the returned buffer.
```

Comment:
add: or -1 on error, in which case the callback implemention must provide a human readable error string in `std::string &errstr`.

change return type to ssize_t
add errstr parameter

Reply:
Again who should allocate this in the C layer?  In C++ we have a std::string which the callback can set but in the C layer this needs to be allocated.  

As for the ssize_t it is not defined so not sure that we want to define it in C++,  It is defined in rdkafka.h but do we want to duplicate that?  What is wrong with size_t in this case?